### ISSUE TITLE
Sorry, we're not able to verify your SSH key: sorry, unrecognized public key type

### ISSUE THREAD

**Issue by MrGraversen** [gogits/gogs#1877]
I'm trying to add an SSH key to my account, but every time I try to do so, it fails with the error message in the title.

The SSH key is generated using puttygen, it's the following format:

---- BEGIN SSH2 PUBLIC KEY ----
Comment: "martin-desktop"
AAAAB3NzaC1yc2EAAAABJQAAAgEAhg6gTei9vbPHhSpAy5jklRM6EXlH/HJtZd59
NeyQFDiqbkZ5yP2mBCVRPGah/XDDIflA7zDb7md+yFOwlqPCsUtL6WP8GDjLY6HD
YEaJIStdbGyxZkhMPnaD6LsCjemWradXs1KpznpIHc0kiGpz1FPA1a/2WmvdGfWg
He41ALufOxaqo59l6npo7Oo3+Cg6HQuVFL5IAGmnObMN8RfWhqUbMgz0hDxLsf8E
N8QFlcprbMVwsJiEI1y5W5AJfpw5OCS4j0iJsL9PiWs2+bb6XrpvvjfKeGV0hjL5
plYgAhXCjpXcz8alHUHAoMuajbujCzQqgzQszB5y2CqYH23y8q35NE+v5/ulE/gT
icTnzVPDO0wdijykvqo9tXo7QxISoUFVglCTatnjPA/tibHPrduAuKOyN9pamgzR
aDl771Y2nO0dckHmS0pLCdFnZJaGGupvDoak4xIuTBm1WA3y91rO3jw3FJit/onN
uFlXsddgXv/n1m8WN8AE3ax2K4d5G2ASllj+WRjZRJTLLMQY+gc1xuX29b+xDYpi
mwKD69isUPVPSLl4wnG67xHOyvnnDPGG8VF2GHUvYc1DiMo9xyt1q3wQFvDUOkSl
RpDQx8s2oAt/83vEysOlbVbh6zdqPWZw9RNuypHnctkce2dylrJjj1LCDCxpVDVi
kr8VcVM=
---- END SSH2 PUBLIC KEY ----

**Comment by JohnMaguire** [comment]
Hi. Please try this:

```
AAAAB3NzaC1yc2EAAAABJQAAAgEAhg6gTei9vbPHhSpAy5jklRM6EXlH/HJtZd59NeyQFDiqbkZ5yP2mBCVRPGah/XDDIflA7zDb7md+yFOwlqPCsUtL6WP8GDjLY6HDYEaJIStdbGyxZkhMPnaD6LsCjemWradXs1KpznpIHc0kiGpz1FPA1a/2WmvdGfWgHe41ALufOxaqo59l6npo7Oo3+Cg6HQuVFL5IAGmnObMN8RfWhqUbMgz0hDxLsf8EN8QFlcprbMVwsJiEI1y5W5AJfpw5OCS4j0iJsL9PiWs2+bb6XrpvvjfKeGV0hjL5plYgAhXCjpXcz8alHUHAoMuajbujCzQqgzQszB5y2CqYH23y8q35NE+v5/ulE/gTicTnzVPDO0wdijykvqo9tXo7QxISoUFVglCTatnjPA/tibHPrduAuKOyN9pamgzRaDl771Y2nO0dckHmS0pLCdFnZJaGGupvDoak4xIuTBm1WA3y91rO3jw3FJit/onNuFlXsddgXv/n1m8WN8AE3ax2K4d5G2ASllj+WRjZRJTLLMQY+gc1xuX29b+xDYpimwKD69isUPVPSLl4wnG67xHOyvnnDPGG8VF2GHUvYc1DiMo9xyt1q3wQFvDUOkSlRpDQx8s2oAt/83vEysOlbVbh6zdqPWZw9RNuypHnctkce2dylrJjj1LCDCxpVDVikr8VcVM= martin-desktop
```

Your key is in an uncommon format.

**Comment by JohnMaguire** [comment]
Hi. Please try the following:

```
ssh-rsa AAAAB3NzaC1yc2EAAAABJQAAAgEAhg6gTei9vbPHhSpAy5jklRM6EXlH/HJtZd59NeyQFDiqbkZ5yP2mBCVRPGah/XDDIflA7zDb7md+yFOwlqPCsUtL6WP8GDjLY6HDYEaJIStdbGyxZkhMPnaD6LsCjemWradXs1KpznpIHc0kiGpz1FPA1a/2WmvdGfWgHe41ALufOxaqo59l6npo7Oo3+Cg6HQuVFL5IAGmnObMN8RfWhqUbMgz0hDxLsf8EN8QFlcprbMVwsJiEI1y5W5AJfpw5OCS4j0iJsL9PiWs2+bb6XrpvvjfKeGV0hjL5plYgAhXCjpXcz8alHUHAoMuajbujCzQqgzQszB5y2CqYH23y8q35NE+v5/ulE/gTicTnzVPDO0wdijykvqo9tXo7QxISoUFVglCTatnjPA/tibHPrduAuKOyN9pamgzRaDl771Y2nO0dckHmS0pLCdFnZJaGGupvDoak4xIuTBm1WA3y91rO3jw3FJit/onNuFlXsddgXv/n1m8WN8AE3ax2K4d5G2ASllj+WRjZRJTLLMQY+gc1xuX29b+xDYpimwKD69isUPVPSLl4wnG67xHOyvnnDPGG8VF2GHUvYc1DiMo9xyt1q3wQFvDUOkSlRpDQx8s2oAt/83vEysOlbVbh6zdqPWZw9RNuypHnctkce2dylrJjj1LCDCxpVDVikr8VcVM= martin-desktop
```

Your key is in SSH2 public key format, rather than the OpenSSH format which is expected. I have [converted](https://burnz.wordpress.com/2007/12/14/ssh-convert-openssh-to-ssh2-and-vise-versa/) the key for you.

I'm not sure if there are plans to support the SSH2 format.

**Comment by Unknwon** [comment]
Maybe will be supported but I have no idea about how this format should be added to `~/.ssh/authorized_keys` file, any idea?

**Comment by matt-h** [comment]
It would need to be reformatted before being put into `~/.ssh/authorized_key`

**Comment by Unknwon** [comment]
@matt-h thanks, reformatted to SSH1 format?

**Comment by JohnMaguire** [comment]
Reformatted to OpenSSH format.

**Comment by JohnMaguire** [comment]
`ssh-keygen -i -f <ssh2_keyfile>` should do the trick.

**Comment by Unknwon** [comment]
Thanks @JohnMaguire .

**Comment by TitinDev** [comment]
no support ssh key ends :(

**Comment by Unknwon** [comment]
@TitinDev can you rephrase your comment 😅

**Comment by TitinDev** [comment]
no support ssh key "ecdsa" (sorry :D)

**Comment by Unknwon** [comment]
I think it is supported, just need to upgrade your OpenSSH version (but not sure).

**Comment by MrGraversen** [comment]
Hello everyone,

Thanks for the replies. I've been busy; sorry I didn't get back sooner.

 @JohnMaguire I pasted the key you kindly provided in your comment, however, it is rejected with the following message:

`Sorry, we're not able to verify your SSH key: sorry, unrecognized public key type`

**Comment by JohnMaguire** [comment]
@MrGraversen Is that the exact message? Interesting that it doesn't specify what key type it detected as it does for example here: https://github.com/gogits/gogs/issues/1880

At this point, @Unknwon could probably help more. I'm fairly certain that it was converted properly, but I don't have a great way to test that. :(

**Comment by Unknwon** [comment]
@JohnMaguire 

> @MrGraversen Is that the exact message? Interesting that it doesn't specify what key type it detected as it does for example here: #1880

That is old error message, the latest release does show which type it detects. https://github.com/gogits/gogs/blob/master/models%2Fpublickey.go#L222

**Comment by MrGraversen** [comment]
Hello everyone, I updated to the latest version (0.7.24 beta), and it now adds the SSH key fine as per what @JohnMaguire posted.

Thanks for the help.

**Comment by Unknwon** [comment]
@MrGraversen :+1:
